### PR TITLE
Allow DG0 on a point

### LINKED ISSUE #209: Support DG 0 elements for points
To be able to support point cloud data, similar to what was presented by: https://fenics2021.com/talks/nixon-hill.html
we need to be able to make DG-0 elements using basix (which can then be used to make a point-cloud mesh in dolfinx).

### PR DESCRIPTION
**Pull request by mscroggs**
Fixes #209.

### CODE REVIEW

File: cpp/basix/cell.cpp [hunk 1]
```diff
@@ -16,6 +16,8 @@ xt::xtensor<double, 2> cell::geometry(cell::type celltype)
 {
   switch (celltype)
   {
+  case cell::type::point:
+    return xt::xtensor<double, 2>({{}});
```
Comment (garth-wells) [inline cpp/basix/cell.cpp:20]: Should this be
```
return xt::xtensor<double, 2>({{0.0}});
```
?

File: cpp/basix/cell.cpp [hunk 2]
```diff
@@ -59,6 +61,11 @@ std::vector<std::vector<std::vector<int>>> cell::topology(cell::type celltype)
 
   switch (celltype)
   {
+  case cell::type::point:
+    t.resize(1);
+    // Vertices
+    t[0] = {{}};
```
Comment (garth-wells) [inline cpp/basix/cell.cpp:67]: Should this be `t[0] = {{0}};`?

**Review by garth-wells — COMMENTED** [review]
Few queries inline on point handling.

File: cpp/basix/cell.cpp (hunk 1 above)
Comment (mscroggs) [reply cpp/basix/cell.cpp:20]: Point has 0 topological dimension so I thought should have coordinates of length 0. But I'm not certain this is sensible

Comment (mscroggs) [reply cpp/basix/cell.cpp]: Tests pass either way, so I'm adding the 0.0 to make it length 1. Can always revise it once we have an example in DOLFINx that actually uses this functionality

**Pull request merged**